src/schematools/contrib/django/factories.py

```diff
@@ -109,26 +101,24 @@ def is_loose_relation(cls, relation, dataset, field):
         return not destination_type_set <= source_type_set
 
     @classmethod
-    def fetch_maker(cls, dataset, field):
+    def fetch_maker(cls, dataset, field: DatasetFieldSchema):
         # determine type of relation (FKLoose, FK, M2M, LooseM2M)
         if field.relation:
-            if cls.is_loose_relation(field.relation, dataset, field):
+            if cls.is_loose_relation(field):
                 return LooseFKRelationMaker
             else:
                 return FKRelationMaker
         elif field.nm_relation:
-            if cls.is_loose_relation(field.nm_relation, dataset, field):
+            if cls.is_loose_relation(field):
                 return LooseM2MRelationMaker
             else:
                 return M2MRelationMaker
         else:
             return None  # To signal this is not a relation
 
     def _make_related_classname(self, relation):
-        related_dataset, related_table = [
-            to_snake_case(part) for part in self.fetch_relation_parts(relation)
-        ]
-        return f"{related_dataset}.{related_table}"
+        related_dataset, related_table, *_ = relation.split(":")
+        return f"{related_dataset}.{to_snake_case(related_table)}"
```

Comment:
is de snake_case op related_dataset hier niet meer nodig?

Reply:
dat heb ik gecheckt, en dat wordt ook niet toegepast bij het genereren van de *app_label*, dus daarom niet doorgevoerd.